pkg/cri/resource-manager/control/cri/cri.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2019 Intel Corporation. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cri
+
+import (
+	"fmt"
+
+	"github.com/intel/cri-resource-manager/pkg/cri/client"
+	"github.com/intel/cri-resource-manager/pkg/cri/resource-manager/cache"
+	"github.com/intel/cri-resource-manager/pkg/cri/resource-manager/control"
+	criapi "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+
+	logger "github.com/intel/cri-resource-manager/pkg/log"
+)
+
+const (
+	// CRIController is the name of this controller.
+	CRIController = cache.CRI
+)
+
+// crictl encapsulated the runtime state of our CRI enforcement/controller.
+type crictl struct {
+	cache  cache.Cache
+	client client.Client
+}
+
+// Make sure crictl implements the control.Controller interface.
+var _ control.Controller = &crictl{}
```

Comment:
Is there a good reason for these?

Reply:
Same as above. Idiomatic pattern I used.

Reply:
Removed these from rdt, blockio, and the cri controller.